### PR TITLE
feat: add calc progress number

### PR DESCRIPTION
**Pull request by HuberTRoy**
## Description

Calculate progress number.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: packages/react-hooks/src/utils.ts [hunk 1]
```diff
@@ -29,6 +29,20 @@ export function mapAsync<O, T>(scope: (t: T) => O, data: AsyncData<T>): AsyncDat
   return { ...data, data: data.data ? scope(data.data) : undefined };
 }
 
+export const deploymentProgessNumber = ({
```
Comment (mzxyz) [inline packages/react-hooks/src/utils.ts:32]: [suggestion] better function name: `indexingProgress`

Comment (HuberTRoy) [reply packages/react-hooks/src/utils.ts]: Yes. better than now. fixed~.

**Comment by mzxyz** [comment]
Just found this function is in `react-hook`, recommend to move to `network-clients` as `react-hook` can only be consumed by frontend app.

**Comment by HuberTRoy** [comment]
@mzxyz 

Ok.

and I also want to adjust the judge order of `renderAsync`.

now:
```
  if (data.data !== undefined) {
    try {
      return handlers.data(data.data, data);
    } catch (e) {
      // TODO not sure this is desired behaviour
      return handlers.error(e as Error);
    }
  } else if (data.error) {
    return handlers.error(data.error);
  } else if (data.loading) {
    return handlers.loading();
  }
```
to:

```
if (data.error) {
    return handlers.error(data.error);
  } else if (data.loading) {
    return handlers.loading();
  } else if (data.data !== undefined) {
    try {
      return handlers.data(data.data, data);
    } catch (e) {
      // TODO not sure this is desired behaviour
      return handlers.error(e as Error);
    }
  }
```

What do you think? It's a breaking change.

**Comment by mzxyz** [comment]
> @mzxyz
> 
> Ok.
> 
> and I also want to adjust the judge order of `renderAsync`.
> 
> now:
> 
> ```
>   if (data.data !== undefined) {
>     try {
>       return handlers.data(data.data, data);
>     } catch (e) {
>       // TODO not sure this is desired behaviour
>       return handlers.error(e as Error);
>     }
>   } else if (data.error) {
>     return handlers.error(data.error);
>   } else if (data.loading) {
>     return handlers.loading();
>   }
> ```
> 
> to:
> 
> ```
> if (data.error) {
>     return handlers.error(data.error);
>   } else if (data.loading) {
>     return handlers.loading();
>   } else if (data.data !== undefined) {
>     try {
>       return handlers.data(data.data, data);
>     } catch (e) {
>       // TODO not sure this is desired behaviour
>       return handlers.error(e as Error);
>     }
>   }
> ```
> 
> What do you think? It's a breaking change.

This is looking good to me, as the react-hook only used by explore, make sure explore resolve the compatible issues when upgrade to the new version

**Comment by HuberTRoy** [comment]
@mzxyz ok~, if only used in explorer then no breaking change.

**Review by mzxyz — APPROVED** [review]

**Pull request merged**